mongo/document/document.go

```diff
@@ -142,25 +136,29 @@ func WhereSortUnscoped(ctx context.Context, collectionName string, query bson.M,
 		query = bson.M{}
 	}
 
-	err := c.Find(query).Sort(sortFields...).All(data)
+	fields := make([]string, len(sortFields))
+	for i, f := range sortFields {
+		fields[i] = string(f)
+	}
+	err := c.Find(query).Sort(fields...).All(data)
 	if err != nil {
 		return fmt.Errorf("fail to query mongo %v: %v", query, err)
 	}
 	return nil
 }
 
-func WhereSort(ctx context.Context, collectionName string, query bson.M, data interface{}, sortFields ...string) error {
+func WhereSort(ctx context.Context, collectionName string, query bson.M, data interface{}, sortFields ...SortField) error {
```

Comment:
This is not needed anymore, we could simply use the Where method

Reply:
Yes true, thanks

Reply:
updated